### ISSUE TITLE
ERR DEPTH_ZERO_SELF_SIGNED_CERT

### ISSUE THREAD

**Issue by rmwiseman** [caputomarcos/node-red-contrib-oauth2#44]
Everything was working on 4.0.1 but when I upgraded to 4.1.1, I suddenly got this error.  Literally nothing else changed, just upgraded to 4.1.1.

I spotted the issue because I had two Node-RED instances with the same configuration, but one worked and the other produced the `ERR DEPTH_ZERO_SELF_SIGNED_CERT` error all the time.  In desperation I checked the OAuth2 node versions, and the broken one was 4.1.1.  The working one was 4.0.1 and I chanced upgrading to 4.1.1 and immediately it produced the same error.

**Comment by caputomarcos** [comment]
Hi @rmwiseman Thanks for getting back!

answering => `https://github.com/caputomarcos/node-red-contrib-oauth2/pull/46#issuecomment-1460273187`

It's possible that the error log is somewhere else, depending on how you have Node-RED set up. One thing you could try is running Node-RED with the "--verbose" flag, which enables verbose output and could provide more information about the error. You can do this by running the command "node-red --verbose" from the command line. Let me know if that helps!

**Comment by rmwiseman** [comment]
> running the command "node-red --verbose" from the command line

Hi @caputomarcos, I've done as you suggested.  I installed v4.1.3 again, restarted Node-RED to include the `--verbose` flag, reloaded my browser, confirmed that the palette shows v4.1.3.  Now the OAuth2 node shows the error again.

Unfortunately, I've looked through all the output, and there is literally no mention of OAuth2 there at all.  No errors except my own, unrelated ones.  I reverted to 4.0.1 again and all is working again.

**Comment by rmwiseman** [comment]
@caputomarcos Hi again, is there anything else I can try to get to the bottom of why the fix doesn't work for me?

**Comment by caputomarcos** [comment]
Hi @rmwiseman,

It seems that the issue you are facing with the bugfix might be related to certificate validation. To disable certificate validation, you can try setting the "rejectUnauthorized" option to false when making HTTPS requests. This might fix the issue for you.

Here's an example of how you can do this in your code:
```js
axios.post(options.url, options.form, {
  headers: options.headers,
  httpsAgent: node.rejectUnauthorized ? new https.Agent({ rejectUnauthorized: true }) : new https.Agent({ rejectUnauthorized: false }),
})
```
Alternatively, you can wait for the next version (4.1.5) where this issue will be fixed.

Let me know if this helps or if you have any further questions!

**Comment by caputomarcos** [comment]
@rmwiseman 

The new version is available:  [Updated node-red-contrib-oauth2 (4.1.5) ](https://flows.nodered.org/node/node-red-contrib-oauth2)
```bash
npm notice Publishing to https://registry.npmjs.org/ with tag latest and default access
+ node-red-contrib-oauth2@4.1.5
```

with kind regards,

**Comment by rmwiseman** [comment]
Thanks @caputomarcos.  My code doesn't call `axios.post` directly - it's all via Node-RED nodes, so I don't know where I would use that snippet.  I am using a `tls-config` node that doesn't verify the server certificate, if that helps.

Will the 4.1.5 fix mean that I don't have to make any changes, and it will just work as it is?  I assume 4.1.5 won't blindly allow any certificate that can't be verified?  Just curious how that will work, whether there will be a warning that the certificate couldn't be verified, or whether there will be a tick box to say not to verify the certificate.

**Comment by rmwiseman** [comment]
> The new version is available: [Updated node-red-contrib-oauth2 (4.1.5) ](https://flows.nodered.org/node/node-red-contrib-oauth2)

Thanks again @caputomarcos.  I don't want to speak too soon, but that appears to have fixed my issue! :smile:

Many thanks for your effort in getting to the bottom of this.
